tests/test_pytypes.cpp

```diff
@@ -923,4 +926,41 @@ TEST_SUBMODULE(pytypes, m) {
 #else
     m.attr("defined_PYBIND11_TYPING_H_HAS_STRING_LITERAL") = false;
 #endif
+
+#if defined(PYBIND11_HAS_RANGES) // test_ranges
+    m.attr("defined_PYBIND11_HAS_RANGES") = true;
+    m.def("iterator_default_initialization", []() {
+        using TupleIterator = decltype(py::tuple{}.begin());
+        using ListIterator = decltype(py::list{}.begin());
+        using DictIterator = decltype(py::dict{}.begin());
+        return py::bool_(
+            TupleIterator{} == TupleIterator{} && ListIterator{} == ListIterator{}
+            && DictIterator{}
+                   == DictIterator{}); // value initialization result must compared as true
+    });
+
+    m.def("transform_tuple_plus_one", [](py::tuple &tpl) {
+        static_assert(std::ranges::random_access_range<py::tuple>);
+        for (auto it : tpl | std::views::transform([](auto &o) { return py::cast<int>(o) + 1; })) {
+            py::print(it);
```

Comment:
Using `py::print` isn't very test friendly. Better build and return a `py::list`.